### PR TITLE
Minor changes to cb config

### PR DESCRIPTION
**Pull request by asbjrnmunk**
Two minor changes:

1. I need to be able to configure a lot of the hard coded parameter values used in Yucca. This allows this and leaves the old values as default values.
2. Cleanup function definitions: add missing types and only define default parameter values in a single place.
3. Add lr monitor

### CODE REVIEW

File: yucca/training/configuration/configure_callbacks.py [hunk 1]
```diff
@@ -55,53 +72,77 @@ def get_loggers(
 
     loggers = [
         YuccaLogger(
-            disable_logging=disable_logging,
+            disable_logging=not enable_logging,
```
Comment (Sllambias) [inline yucca/training/configuration/configure_callbacks.py:75]: should this be enable_logging=enable_logging now for consistency?

**Review by Sllambias — COMMENTED** [review]
Just a few comments/questions:
1. Is there ever a case where write_predictions will be false? It's only active during inference and I don't know if we will ever want it off
2. Why do you need to log the lr? (i'm not against it, I'm just curious)
3. Would you like to be able to turn off storing the best ckpt? (also mostly curiousity)

File: yucca/training/configuration/configure_callbacks.py (hunk 1 above)
Comment (asbjrnmunk) [reply yucca/training/configuration/configure_callbacks.py:75]: Yeah I guess, but we can always change it later. I just noticed that we every else did `if not disable_logging`, so might as well not in once.

File: yucca/training/configuration/configure_callbacks.py [hunk 2]
```diff
@@ -55,53 +72,77 @@ def get_loggers(
...
         )
+
     return loggers
 
 
-def get_callbacks(checkpoint_interval, prediction_output_dir: str = None, save_softmax: bool = False):
-    best_ckpt = ModelCheckpoint(monitor="val_loss", mode="min", save_top_k=1, filename="best", enable_version_counter=False)
+def get_callbacks(
+    interval_ckpt_epochs: int,
+    latest_ckpt_epochs: int,
```
[the remark this answers is not in the document]
Comment (asbjrnmunk) [reply yucca/training/configuration/configure_callbacks.py:99]: How about `interval_ckpt_every_n_epochs` and `latest_ckpt_every_n_epochs`?

**Comment by asbjrnmunk** [comment]
>Is there ever a case where write_predictions will be false? It's only active during inference and I don't know if we will ever want it off

When doing pre-training it will break the model if it is every run, since we are not using logits but just predicting the signal in a single channel through a sigmoid.

>Why do you need to log the lr? (i'm not against it, I'm just curious)

It works really well in wandb to understand your training curves. Also, since im going to try different learning curves it allows me to quickly see if something is off (which it was in my last run). 

>Would you like to be able to turn off storing the best ckpt? (also mostly curiousity)
Yes, since I am never going to use it. This is also implemented in the PR:
https://github.com/Sllambias/yucca/blob/9ddd813ed2e882c3e8804d634abc252dab839649/yucca/training/configuration/configure_callbacks.py#L118-L122

File: yucca/training/configuration/configure_callbacks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sllambias) [reply yucca/training/configuration/configure_callbacks.py:99]: works for me

File: yucca/training/configuration/configure_callbacks.py (hunk 1 above)
Comment (Sllambias) [reply yucca/training/configuration/configure_callbacks.py:75]: I think it would be nice to change the variable everywhere if we start calling it something else

**Review by Sllambias — APPROVED** [review]
lgtm!
giphy
